sympy/sets/ordinals.py

```diff
@@ -0,0 +1,152 @@
+from sympy.core import Basic
+
+def reverse_bisect_right(a, x, lo=0, hi=None):
+    """Return the index where to insert item x in list a, assuming a is sorted in descending order
+
+    Essentially, the function returns number of elements in a which are >= than x.
+    >>> a = [8, 6, 5, 4, 2]
+    >>> reverse_bisect_right(a, 5)
+    3
+    >>> a[:reverse_bisect_right(a, 5)]
+    [8, 6, 5]
+    """
+    if lo < 0:
+        raise ValueError('lo must be non-negative')
+    if hi is None:
+        hi = len(a)
+    while lo < hi:
+        mid = (lo+hi)//2
+        if x > a[mid]: hi = mid
+        else: lo = mid+1
+    return lo
+
+class Ordinal(Basic):
+    """
+    Represents ordinals in Cantor normal form.
+
+    Internally, this class is just a list of lists (known
+    as "terms").
+    In single terms [a,b,c], a represents index of ordinal number
```

Comment:
I think that the 'base' of ordinals in Cantor normal form is always `omega`. Hence the index `a` should be 0 and could be omitted from input. (`omega` itself would be represented by `b, c = 1, 1`.)